common/ledger/blockledger/fileledger/factory.go

```diff
@@ -55,6 +55,13 @@ func (f *fileLedgerFactory) Remove(channelID string) error {
 	f.mutex.Lock()
 	defer f.mutex.Unlock()
 
+	// check cache for open blockstore and, if one exists,
+	// shut it down in order to avoid resource contention
+	ledger, ok := f.ledgers[channelID]
+	if ok {
+		ledger.blockStore.Shutdown()
+	}
```

Comment:
I assume that Remove can be called without first calling GetOrCreate. In that case, since blockstore is not opened yet, it should not be a problem.

Reply:
Right, that was my thinking here. I was hitting an error in the unit tests with existing ledgers on the filesystem since GetOrCreate hadn't been called and thus they were in the fileLedgerFactory's ledgers cache.